src/client.rs

```diff
@@ -1922,6 +2000,29 @@ mod test {
         }
     }
 
+    #[tokio::test]
+    async fn test_list_tags() {
+        let reference = Reference::try_from("docker.io/nginx").expect("Can't fail in theory");
+        let auth = RegistryAuth::Anonymous;
+
+        let mut client = Client::default();
+
+        let response = client
+            .list_tags(&reference, &auth, Some(5), Some("1.13.10"))
+            .await
+            .expect("Cannot list Tags");
+        assert_eq!(
+            response.tags,
+            vec![
+                "1.13.10-alpine",
+                "1.13.10-alpine-perl",
+                "1.13.10-perl",
+                "1.13.11",
+                "1.13.11-alpine"
+            ]
```

Comment:
so i just implemented the new test but i found that the registry image used for local testing is does not return oci compliant responses. 

the tags are not in lexical order and the query parameters seem to be ignored. i also sanity checked with a http client.

could you confirm?

Reply:
This looks strange, the tests are using the latest stable release of the [registry](https://hub.docker.com/_/registry) image, which is based on the [`distribution/distribution`](https://github.com/distribution/distribution) project.

This is the reference implementation of the OCI specification, looking into their open issues I found [this epic](https://github.com/distribution/distribution/issues/3203) that tracks the OCI compliance.

Can you report the issue(s) inside of the distribution project and link them back here? Thanks a lot! :heart:

Reply:
they have [this PR](https://github.com/distribution/distribution/pull/3143) which fixes those things and is already on main.

but from the looks of it it will only come on 3.0.0 which seems pretty far off according to the [milestone](https://github.com/distribution/distribution/milestone/22)

Reply:
@flavio what just came to my mind: the dockerhub rate limiting is image pulls/hour and i wouldn't think that content discovery counts towards that. so i don't think we would have any rate limiting issues in this case when testing against dockerhub.

just a thought

Reply:
@rbbl-dev, sorry about the delay...

I think I found a simpler approach to solve this limitation, take a look at c4c6e2dcb67f5eb2ed9bc836e0504a537b9d6d5b - which I just pushed to your branch.

If you like I think we can move forward and get this PR merged

Reply:
this approach would not test the url parameters at all. 

the comment about the issues with registry v2 is also not really complete since it does not mention the url params.

so i'm on the fence in terms of the usefulness of the test but its your project so you can merge it in if you want to.

Reply:
Sorry for the late delay. I removed by previous attempt to fix the issue because it made the test useless.

Instead, I've found that `distribution/distribution:edge` is a container image that is regularly built from the `main` branch. I've changed your unit test code to consume this image, instead of the `v2` one. The list tags api works as expected with this image, I've tested it by doing manual GET requests.

I only had to fix the assertion you made because it was looking for the wrong values.

Can you double check the fixes done with 23aae2add450364303766b22581986a85ac19aa7 ?

Reply:
looks fine to me.

tho i now seem to have broken the windows pipeline even though i just made a little cleanup squash